components/Cards/ResourceCard/__tests__/ResourceCard.test.tsx

```diff
@@ -12,12 +11,13 @@ describe('ResourceCard', () => {
 
   it('fires appropriate method when upvote button clicked', () => {
     const handleVoteMock = vi.fn();
-
     const component = render(<ResourceCard {...requiredProps} handleVote={handleVoteMock} />);
-
     const UpvoteButton = component.queryByTestId(UPVOTE_BUTTON);
+
     expect(handleVoteMock).not.toHaveBeenCalled();
-    fireEvent.click(UpvoteButton);
+
+    UpvoteButton && fireEvent.click(UpvoteButton);
```

Comment:
```suggestion
   fireEvent.click(UpvoteButton!);
```

we never wanna do conditional testing, but you can use the non-null assertion operator.

Reply:
Super thanks for the heads up - just got to read up on that. There is so much to types and Typescript that I am still missing out on lol.